todo_app/data/mongo_db_manager.py

```diff
@@ -0,0 +1,125 @@
+        if os.getenv('USE_MOCK_MONGO') == 'True':
+            self.client = mongomock.MongoClient(
+                f"mongodb://{os.getenv('MONGO_USER_NAME')}"
+            )
+        else:
+            self.client = MongoClient(
+                f"mongodb+srv://{os.getenv('MONGO_USER_NAME')}:"
+                f"{os.getenv('MONGO_PASSWORD')}@"
+                f"{os.getenv('MONGO_HOST')}/"
+                f"{os.getenv('MONGO_DEFAULT_DATABASE')}?w=majority"
+            )
+        self.db = self.client[os.getenv("MONGO_DEFAULT_DATABASE")]
+
+    def get_collection(self, collection_name):
+        """
+        Retrieves a collection from the database.
+
+        Args:
+            collection_name (str): The name of the collection to retrieve.
+
+        Returns:
+            Collection: The MongoDB collection object.
+        """
+        return self.db[collection_name]
+
+    def insert_document(self, collection_name, document):
+        """
+        Inserts a document into a specified collection.
+
+        Args:
+            collection_name (str): The name of the collection.
+            document (dict): The document to insert.
+
+        Returns:
+            ObjectId: The ID of the inserted document.
+        """
+        collection = self.get_collection(collection_name)
+        return collection.insert_one(document).inserted_id
+
+    def find_document(self, collection_name, query):
+        """
+        Finds a single document in a collection based on a query.
+
+        Args:
+            collection_name (str): The name of the collection.
+            query (dict): The query criteria to find the document.
+
+        Returns:
+            dict: The first document matching the query.
+        """
+        collection = self.get_collection(collection_name)
+        return collection.find_one(query)
+
+    def update_document(self, collection_name, query, new_values):
+        """
+        Updates a single document in a collection based on a query.
+
+        Args:
+            collection_name (str): The name of the collection.
+            query (dict): The query criteria to find the document.
+            new_values (dict): The new values to update the document with.
+
+        Returns:
+            UpdateResult: The result of the update operation.
+        """
+        collection = self.get_collection(collection_name)
+        return collection.update_one(query, {"$set": new_values})
+
+    def delete_document(self, collection_name, query):
+        """
+        Deletes a single document from a collection based on a query.
+
+        Args:
+            collection_name (str): The name of the collection.
+            query (dict): The query criteria to find the document.
+
+        Returns:
+            DeleteResult: The result of the delete operation.
+        """
+        collection = self.get_collection(collection_name)
+        return collection.delete_one(query)
+
+    def list_documents(self, collection_name):
```

Comment:
I'm not sure all of these methods are particularly valuable at the moment - it's adding a wrapper between the calling code and the Mongo functions, but asking them to know what collection name to use for example suggests it's not very abstract, so is there any reason not to give them the collection itself?

(Definitely a thought experiment rather than a suggestion to change, there's nothing wrong with this!)